### ISSUE TITLE
Problem with training of mobilenet and densenet models

### ISSUE THREAD

**Issue by Zhilin27** [fizyr/keras-retinanet#767]
I have trained the the resnet models without any problem. But I got the following when I try to train the mobilenet or densenet models:

> retinanet-train --backbone=mobilenet128_1.0 csv preprocessing/data/ImageSets/Main/train_keras.csv preprocessing/data/ImageSets/Main/class_keras.csv
Using TensorFlow backend.
/home//.local/lib/python3.6/site-packages/keras_retinanet/bin/train.py:352: UserWarning: Using experimental backbone mobilenet128_1.0. Only resnet50 has been properly tested.
  warnings.warn('Using experimental backbone {}. Only resnet50 has been properly tested.'.format(parsed_args.backbone))
2018-10-27 21:36:49.409530: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:897] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2018-10-27 21:36:49.410016: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1405] Found device 0 with properties: 
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.683
pciBusID: 0000:01:00.0
totalMemory: 10.91GiB freeMemory: 10.74GiB
2018-10-27 21:36:49.410028: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1484] Adding visible gpu devices: 0
2018-10-27 21:36:50.114417: I tensorflow/core/common_runtime/gpu/gpu_device.cc:965] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-10-27 21:36:50.114444: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971]      0 
2018-10-27 21:36:50.114451: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] 0:   N 
2018-10-27 21:36:50.115082: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1097] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10390 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
Creating model, this may take a second...
Traceback (most recent call last):
  File "/home//.local/bin/retinanet-train", line 11, in <module>
    sys.exit(main())
  File "/home//.local/lib/python3.6/site-packages/keras_retinanet/bin/train.py", line 461, in main
    config=args.config
  File "/home//.local/lib/python3.6/site-packages/keras_retinanet/bin/train.py", line 118, in create_models
    model          = model_with_weights(backbone_retinanet(num_classes, num_anchors=num_anchors, modifier=modifier), weights=weights, skip_mismatch=True)
  File "/home//.local/lib/python3.6/site-packages/keras_retinanet/models/mobilenet.py", line 35, in retinanet
    return mobilenet_retinanet(*args, backbone=self.backbone, **kwargs)
  File "/home//.local/lib/python3.6/site-packages/keras_retinanet/models/mobilenet.py", line 98, in mobilenet_retinanet
    backbone = mobilenet.MobileNet(input_tensor=inputs, alpha=alpha, include_top=False, pooling=None, weights=None)
  File "/home//.local/lib/python3.6/site-packages/keras_applications/mobilenet.py", line 183, in MobileNet
    data_format=backend.image_data_format(),
AttributeError: 'NoneType' object has no attribute 'image_data_format'

I used keras 2.2.4 and also tried to downgrade to keras 2.2.2, but it didn't solve the problem.

**Comment by hgaiser** [comment]
Are you using the latest keras-retinanet? Also, what tensorflow are you using? Try using at least 1.8.

**Comment by Zhilin27** [comment]
> Are you using the latest keras-retinanet? Also, what tensorflow are you using? Try using at least 1.8.

Yes, the latest keras-retinanet and tensorflow 1.10.0.

**Comment by yuyijie1995** [comment]
Using TensorFlow backend.
Traceback (most recent call last):
  File "D:/postgraduateworking/suanfa/keras-retinanet/keras_retinanet/bin/convert_model.py", line 59, in <module>
    main()
  File "D:/postgraduateworking/suanfa/keras-retinanet/keras_retinanet/bin/convert_model.py", line 52, in main
    model = models.load_model(args.model_in, convert=True, backbone_name=args.backbone, nms=args.nms)
  File "D:/postgraduateworking/suanfa/keras-retinanet/keras_retinanet/bin\..\..\keras_retinanet\models\__init__.py", line 83, in load_model
    model = keras.models.load_model(filepath, custom_objects=backbone(backbone_name).custom_objects)
  File "D:\anaconda\envs\tensorflow\lib\site-packages\keras\engine\saving.py", line 260, in load_model
    model = model_from_config(model_config, custom_objects=custom_objects)
  File "D:\anaconda\envs\tensorflow\lib\site-packages\keras\engine\saving.py", line 334, in model_from_config
    return deserialize(config, custom_objects=custom_objects)
  File "D:\anaconda\envs\tensorflow\lib\site-packages\keras\layers\__init__.py", line 55, in deserialize
    printable_module_name='layer')
  File "D:\anaconda\envs\tensorflow\lib\site-packages\keras\utils\generic_utils.py", line 145, in deserialize_keras_object
    list(custom_objects.items())))
  File "D:\anaconda\envs\tensorflow\lib\site-packages\keras\engine\network.py", line 1017, in from_config
    process_layer(layer_data)
  File "D:\anaconda\envs\tensorflow\lib\site-packages\keras\engine\network.py", line 1003, in process_layer
    custom_objects=custom_objects)
  File "D:\anaconda\envs\tensorflow\lib\site-packages\keras\layers\__init__.py", line 55, in deserialize
    printable_module_name='layer')
  File "D:\anaconda\envs\tensorflow\lib\site-packages\keras\utils\generic_utils.py", line 147, in deserialize_keras_object
    return cls.from_config(config['config'])
  File "D:\anaconda\envs\tensorflow\lib\site-packages\keras\engine\base_layer.py", line 1109, in from_config
    return cls(**config)
TypeError: __init__() missing 1 required positional argument: 'freeze'
I used the mobilenet complete the trainning ,but when I converted the model,this error happened ,can you tell me how to solve it? @hgaiser thanks a lot

**Comment by hgaiser** [comment]
> I used the mobilenet complete the trainning ,but when I converted the model,this error happened ,can you tell me how to solve it? @hgaiser thanks a lot

This is a different issue, I responded in the issue you created.



> Yes, the latest keras-retinanet and tensorflow 1.10.0.

The error you showed, is that from keras 2.2.2? I just tried it with keras 2.2.4 and keras-retinanet 0.5.0, for me it starts training. Make sure all your packages are up to date. I see you're using an installed version of keras-retinanet, make sure it is the updated one or it won't work.

The error is weird though, because it complains that keras.backend is `None`. I think the issue is more likely to be in Keras, than in keras-retinanet.
